pkgs/development/interpreters/python/wrapper.nix

```diff
@@ -42,7 +42,15 @@ let
             if [ -f "$prg" ]; then
               rm -f "$out/bin/$prg"
               if [ -x "$prg" ]; then
-                makeWrapper "$path/bin/$prg" "$out/bin/$prg" --set NIX_PYTHONPREFIX "$out" --set NIX_PYTHONEXECUTABLE ${pythonExecutable} --set NIX_PYTHONPATH ${pythonPath} ${lib.optionalString (!permitUserSite) ''--set PYTHONNOUSERSITE "true"''} ${lib.concatStringsSep " " makeWrapperArgs}
+                if [ -f ".$prg-wrapped" ] && ( cat ".$prg-wrapped" | head -n 1 | grep -q "python" ) ; then
+                  echo "#!${pythonExecutable}" > "$out/bin/$prg"
+                  echo "import os" > "$out/bin/$prg"
+                  echo 'os.environ["NIX_PYTHON_IN_ENV"] = "true"' > "$out/bin/$prg"
+                  cat ".$prg-wrapped" >> "$out/bin/$prg"
```

Comment:
The first 3 echo lines are wrong. You probably meant to do `>>` redirection on all of them instead of `>`. 

For example, building `python3.withPackages (ps: [ ps.gyp ])` currently produces a `bin/gyp` file that looks like this:
```python
os.environ["NIX_PYTHON_IN_ENV"] = "true"
#!/nix/store/rqj8pv1l61jm2szvz8w7k1af54hfq75d-python3-3.12.5/bin/python3.12
# -*- coding: utf-8 -*-
import sys;import site;import functools;import os;_nix_direct = os.getenv("NIX_PYTHON_IN_ENV", None) != "true";sys.argv[0] = '/nix/store/mrrarzbn67cvgzgp1qqf6m1i8kz83861-python3.12-gyp-unstable-2022-04-01/bin/gyp' if _nix_direct else sys.argv[0];functools.reduce(lambda k, p: site.addsitedir(p, k), ['/nix/store/mrrarzbn67cvgzgp1qqf6m1i8kz83861-python3.12-gyp-unstable-2022-04-01/lib/python3.12/site-packages','/nix/store/iy6q6wmwawi9b4kxc7y8vj7yr2gqi9nb-python3.12-six-1.16.0/lib/python3.12/site-packages'], site._init_pathinfo()) if _nix_direct else None;os.environ.pop(NIX_PYTHON_IN_ENV) if not _nix_direct else None;
import re
import sys
from gyp import script_main
if __name__ == "__main__":
    sys.argv[0] = re.sub(r"(-script\.pyw|\.exe)?$", "", sys.argv[0])
    sys.exit(script_main())
```
which doesn't start with a shebang.

Also, even if you used `>>` instead of `>`, it would still generate something like
```python
#!/nix/store/azrly9ksxm0qzyyjv1vbn5fwg970da9a-python3-3.12.5-env/bin/python3.12
import os
os.environ["NIX_PYTHON_IN_ENV"] = "true"
#!/nix/store/rqj8pv1l61jm2szvz8w7k1af54hfq75d-python3-3.12.5/bin/python3.12
# -*- coding: utf-8 -*-
import sys;import site;import functools;import os;_nix_direct = os.getenv("NIX_PYTHON_IN_ENV", None) != "true";sys.argv[0] = '/nix/store/mrrarzbn67cvgzgp1qqf6m1i8kz83861-python3.12-gyp-unstable-2022-04-01/bin/gyp' if _nix_direct else sys.argv[0];functools.reduce(lambda k, p: site.addsitedir(p, k), ['/nix/store/mrrarzbn67cvgzgp1qqf6m1i8kz83861-python3.12-gyp-unstable-2022-04-01/lib/python3.12/site-packages','/nix/store/iy6q6wmwawi9b4kxc7y8vj7yr2gqi9nb-python3.12-six-1.16.0/lib/python3.12/site-packages'], site._init_pathinfo()) if _nix_direct else None;os.environ.pop(NIX_PYTHON_IN_ENV) if not _nix_direct else None;
import re
import sys
from gyp import script_main
if __name__ == "__main__":
    sys.argv[0] = re.sub(r"(-script\.pyw|\.exe)?$", "", sys.argv[0])
    sys.exit(script_main())
```
which is a slightly weird wrapper, because it writes the `NIX_PYTHON_IN_ENV` environment variable only to immediately read from it and pop it.

Is this actually needed? More specifically - are there any cases where `os.environ["NIX_PYTHON_IN_ENV"] = "true"` and the `os.getenv("NIX_PYTHON_IN_ENV", None)` are actually not performed in the same file?

I still think that the approach from the previous PRs that stripped the unneeded wrapper was better.

Reply:
I'm actively trying to avoid having to pattern match on the file in any way shape or form. Simply prepending should not be able to cause any problems. And yes my mistake, don't know how it worked for me. Ill push a fix tomorrow.

Reply:
@RuRo while i agree that removing the wrapper would be better, adding is easier than removing :) and less error prone. This way we don't have to make sure we never remove anything by accident, because we never remove anything.